application/server/server.go

```diff
@@ -401,28 +398,36 @@ func monitorClientProcess(pid int) time.Duration {
 	return time.Since(start)
 }
 
-func shutdown() jrpc2.Handler {
+func shutdown(c *config.Config) jrpc2.Handler {
 	return handler.New(func(ctx context.Context) (any, error) {
-		log.Info().Str("method", "Shutdown").Msg("RECEIVING")
-		defer log.Info().Str("method", "Shutdown").Msg("SENDING")
+		logger := c.Logger().With().Str("method", "Shutdown").Logger()
+		logger.Info().Msg("ENTERING")
+		defer logger.Info().Msg("RETURNING")
 		di.ErrorReporter().FlushErrorReporting()
 
-		disposeProgressListener()
-		di.Notifier().DisposeListener()
-		err := di.Analytics().Shutdown()
-		if err != nil {
-			log.Error().Str("method", "Shutdown").Msg("Failed to shutdown analytics.")
-		}
+		go disposeProgressListener()
+		go di.Notifier().DisposeListener()
+		go func() {
+			err := di.Analytics().Shutdown()
+			if err != nil {
+				logger.Err(err).Msg("Error shutting down analytics")
+			}
+		}()
 		return nil, nil
 	})
 }
 
-func exit(srv *jrpc2.Server) jrpc2.Handler {
+func exit(srv *jrpc2.Server, c *config.Config) jrpc2.Handler {
 	return handler.New(func(_ context.Context) (any, error) {
-		log.Info().Str("method", "Exit").Msg("RECEIVING")
-		log.Info().Msg("Stopping server...")
-		(*srv).Stop()
-		di.ErrorReporter().FlushErrorReporting()
+		logger := c.Logger().With().Str("method", "Exit").Logger()
+		logger.Info().Msg("ENTERING")
+		logger.Info().Msg("Flushing error reporting...")
+		go di.ErrorReporter().FlushErrorReporting()
+		logger.Info().Msg("Stopping server...")
+		go srv.Stop()
+		time.Sleep(time.Second * 1)
+		logger.Info().Msg("Exiting...")
+		os.Exit(0)
```

Comment:
Question: Do we really need to call os.Exit() here? Seems quite a strong move to me. It prevents running an Extension to the end which is not good, as it includes clean up tasks ...

Reply:
I was wondering about that... I'll change it.